### PR TITLE
[ide] VSCode recommended plugins and settings

### PR DESCRIPTION
**Pull request by zach2good**
Should help some people's Lua workflows
(VSCode settings files don't have trailing whitespace and I _hate_ it)

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "recommendations": [
+        "koihik.vscode-lua-format",
+        "jkiviluoto.tws",
+        "sumneko.lua",
+        "wwm.better-align"
+    ]
+}
```
Comment (claywar) [inline .vscode/extensions.json:8]: Newline please! :)

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,441 @@
+{
+    "Lua.workspace.maxPreload": 20000,
+    "Lua.diagnostics.globals": [
```
Comment (claywar) [inline .vscode/settings.json:3]: I wonder, is there a way we could centralize global definitions for linting?  This looks like its following the luacheck CI script, but we might want something easier to maintain in the future (once this list begins to hopefully shrink)

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+    ],
+    "Lua.diagnostics.disable": [
+        "lowercase-global"
+    ],
+    "Lua.hint.enable": true,
+    "Lua.hint.setType": true,
+    "Lua.runtime.version": "LuaJIT",
+    "Lua.workspace.preloadFileSize": 10000,
+    "editor.renderWhitespace": "all"
+}
```
Comment (claywar) [inline .vscode/settings.json:441]: And just one one newline

File: .vscode/settings.json (hunk 2 above)
Comment (zach2good) [reply .vscode/settings.json:3]: One difference between the CI and this is that the CI scraped luautils.cpp for the function names, so it's a little messy

**Pull request merged**